src/components/Timeout/Timeout.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './timeout.css';
+
+export default function Timeout(props) {
+  //track idle time in background
+  //set time in minutes for idle time before modal should appear, and countdown time before auto logout
+  const idleTime = 1;
+  const countdownTime = 5;
+
+  const [idle, setIdle] = useState(false);
+
+  const time = useRef(0);
+
+  function reset() {
+    time.current = 0;
+  }
+
+  useEffect(() => {
+    let interval = setInterval(() => {
+      time.current += 1;
+      if (time.current === idleTime * 60) {
+        setIdle(true);
+      }
+      console.log('idletime', time.current);
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, [idleTime]);
+
+  return (
+    <div onMouseOver={() => reset()} onKeyPress={() => reset()}>
+      {idle && <Modal countdownTime={countdownTime} setIdle={setIdle} />}
+    </div>
+  );
+}
+
+const Modal = ({ countdownTime, setIdle }) => {
+  //modal with countdown clock
+  const [minutes, setMinutes] = useState(Math.floor(countdownTime));
+  const [seconds, setSeconds] = useState((countdownTime * 60) % 60);
```

Comment:
when you switch to using Date instead of Integer, you can use native javascript to get the minutes and seconds from milliseconds. 

check this out in your browser console:
```
const date = new Date()
date.getMinutes()
date.getSeconds()
```

or if you want, dayjs is really good for time related things https://day.js.org/docs/en/get-set/minute#docsNav

Reply:
Yeah I've used Date() before, but in this case, it'd just be taking an extra step on my interval. At the end of each interval, I'd have to increment and compare the Date. There's not exactly a way to tell Javascript to automatically let me know when the Date hits 5 minutes away, right?

Though I do measuring time would be more accurate with Date, since interval is affected by runtime. You see a way to do this without using setInterval? and I prefer not to use a library if it's not necessary.

Reply:
clunking this with the previous comment about using Date() too btw

Reply:
`lastUpdated` replaces the need for counting / incrementing

you can find out how long a user has been idle by subtracting `Date.now` and `lastUpdated`

setInterval is fine, but you can also use recursive setTimeout instead so you don't have to worry about clearing interval